tests/Provider/BaseProviderTest.php

```diff
@@ -16,19 +16,16 @@
 use Gaufrette\Adapter;
 use Gaufrette\File;
 use Gaufrette\Filesystem;
-use Sonata\AdminBundle\Form\FormMapper;
 use Sonata\MediaBundle\CDN\CDNInterface;
 use Sonata\MediaBundle\Generator\IdGenerator;
-use Sonata\MediaBundle\Model\MediaInterface;
 use Sonata\MediaBundle\Provider\BaseProvider;
-use Sonata\MediaBundle\Provider\MediaProviderInterface;
+use Sonata\MediaBundle\Tests\App\Provider\TestProvider;
 use Sonata\MediaBundle\Tests\Entity\Media;
 use Sonata\MediaBundle\Thumbnail\ThumbnailInterface;
-use Symfony\Component\Form\FormBuilderInterface;
 
 class BaseProviderTest extends AbstractProviderTest
 {
-    public function getProvider(): MediaProviderInterface
+    public function getProvider(): TestProvider
```

Comment:
How show I deal with this? this method is abstract and returns an interface, but the concrete classes have more methods and phpstan complains, but if I do this psalm complains on higher levels, and php 7.3 complains too.

Reply:
Which method do you need more than the one in MediaProviderInterface ?

Quick solution would be to keep the `MediaProviderInterface` return type and add `@return TestProvider`, it solve the php 7.3 error. But psalm will complain on higher level I think.

Right solution would be to make the `AbstractProviderTest` using a template
`@phpstan-template P of MediaProviderInterface` and
```
/** @phpstan-return P */
public function getProvider(): MediaProviderInterface;
```

Reply:
I will go for the template solution, seems to be the most accurate

Reply:
Done